### PR TITLE
deps: bump operator-registry version

### PR DESCRIPTION
**Pull request by estroz**
**Description of the change:**
* deps: bump operator-registry
* cmd/operator-sdk/bundle/validate,internal/scorecard/alpha/tests: the NewImageValidator API changed such that the command must create an image registry first before creating a validator.

**Motivation for the change:** This version of operator-registry contains a fix for a bug that causes validation errors when an existing `annotations.yaml` does not exactly match the default annotations generated by `operator-sdk bundle create`.

/cc @camilamacedo86 @jmrodri @hasbro17 @varshaprasad96 

/kind bug

### CODE REVIEW

File: cmd/operator-sdk/bundle/validate.go [hunk 1]
```diff
@@ -23,6 +23,9 @@ import (
```
Comment (camilamacedo86) [inline cmd/operator-sdk/bundle/validate.go:23]: Should it be merged before https://github.com/operator-framework/operator-sdk/pull/3222?

File: cmd/operator-sdk/bundle/validate.go [hunk 2]
```diff
@@ -23,6 +23,9 @@ import (
 
 	apimanifests "github.com/operator-framework/api/pkg/manifests"
 	apierrors "github.com/operator-framework/api/pkg/validation/errors"
+	"github.com/operator-framework/operator-registry/pkg/containertools"
+	registryimage "github.com/operator-framework/operator-registry/pkg/image"
+	"github.com/operator-framework/operator-registry/pkg/image/execregistry"
 	registrybundle "github.com/operator-framework/operator-registry/pkg/lib/bundle"
 	log "github.com/sirupsen/logrus"
```
Comment (camilamacedo86) [inline cmd/operator-sdk/bundle/validate.go:30]: it is missing a fragment file with the bugfix.

File: cmd/operator-sdk/bundle/validate.go (hunk 1 above)
Comment (estroz) [reply cmd/operator-sdk/bundle/validate.go:23]: From #3222's description:
>Blocked by #3221

File: changelog/fragments/3221-bugfix.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+entries:
+  - description: >
+      Bump operator-registry to version containing a fix for a bug that causes validation errors
+      when an existing annotations.yaml does not exactly match the default annotations generated by
+      `operator-sdk bundle create`
```
Comment (camilamacedo86) [inline changelog/fragments/3221-bugfix.yaml:5]:
```suggestion
      Fix error in the `operator-sdk bundle validation` that causes validation errors when an existing `annotations.yaml` does not precisely match the default annotations generated by the `operator-sdk bundle create` command via upgrading the `operator-registry` dependency.

      Bump operator-registry to version containing a fix for a bug that causes validation errors
      when an existing annotations.yaml does not exactly match the default annotations generated by
      `operator-sdk bundle create`
```

File: internal/scorecard/alpha/tests/olm.go [hunk 4]
```diff
@@ -23,6 +23,8 @@ import (
 	apimanifests "github.com/operator-framework/api/pkg/manifests"
 	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
 	apivalidation "github.com/operator-framework/api/pkg/validation"
+	"github.com/operator-framework/operator-registry/pkg/containertools"
+	"github.com/operator-framework/operator-registry/pkg/image/execregistry"
 	registrybundle "github.com/operator-framework/operator-registry/pkg/lib/bundle"
```
Comment (camilamacedo86) [inline internal/scorecard/alpha/tests/olm.go:28]: fix import order

File: changelog/fragments/3221-bugfix.yaml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+entries:
+  - description: >
+      Fix bug in `operator-sdk bundle validation` that causes erroneous validation errors when the number of
+      annotations in an existing `annotations.yaml` does not equal the number of default bundle annotations
+      by upgrading the `operator-registry` dependency.
```
Comment (camilamacedo86) [inline changelog/fragments/3221-bugfix.yaml:5]: Loved 👍

Comment (tengqm) [reply changelog/fragments/3221-bugfix.yaml:5]: Nit. The command should be `operator-sdk bundle validate`.

**Pull request merged**